gen3-client/g3cmd/retry-upload.go

```diff
@@ -79,6 +81,10 @@ func retryUpload(failedLogMap map[string]commonUtils.RetryObject) {
 		log.Printf("Sleep for %.0f seconds\n", GetWaitTime(ro.RetryCount).Seconds())
 		time.Sleep(GetWaitTime(ro.RetryCount)) // exponential wait for retry
 
+		if ro.Bucket !=nil || *ro.Bucket != "" {
```

Comment:
I think there should be a space `!= nil`

Reply:
Also this is a little confusing. I think if Bucket was just a string like GUID this would be a little easier.